src/effects/DtmfGen.cpp

```diff
@@ -386,6 +415,14 @@ EffectDtmf::PopulateOrExchange(ShuttleGui & S, EffectSettingsAccess &access)
    return result;
 }
 
+std::shared_ptr<EffectInstance>
+EffectDtmf::MakeInstance(EffectSettings &settings)
+{
+   // TODO: don't use Effect::mT0 and Effect::mSampleRate, but use an
```

Comment:
When I try to generate a sequence, I get an access violation. I traced the root cause to this point.

When this point is reached, mSampleRate is 0.0 (as defaulted in class `Effect` and never changed then) and that is sent to the created instance.

That, in turn, will make `numSamplesSequence`, `numSamplesTone`, `numSamplesSilence` be all 0 when they are calculated in `EffectDtmf::Instance::ProcessInitialize`, and that in turn will trigger a violation access in EffectDtmf::Instance::ProcessBlock (because the correct indexing of the string containing the sequence depends on those values being right).

Reply:
Yes.

Please see the latest force push, in which the Instance class overrides SetSampleRate().

And now this PR depends on not one but two other pending PRs !

Reply:
I see this has been fixed in later commits, ok